### PR TITLE
mkrom: init at 1.0.1

### PR DESCRIPTION
**Pull request by siraben**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by siraben** [comment]
@GrahamcOfBorg build mkrom

File: pkgs/development/tools/misc/mkrom/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{ fetchFromGitHub, stdenv, cmake, asciidoc, libxslt }:
+
+
+stdenv.mkDerivation rec {
+  pname = "mkrom";
+
```
Comment (danieldk) [inline pkgs/development/tools/misc/mkrom/default.nix:6]:
```suggestion
```

File: pkgs/development/tools/misc/mkrom/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{ fetchFromGitHub, stdenv, cmake, asciidoc, libxslt }:
+
```
Comment (danieldk) [inline pkgs/development/tools/misc/mkrom/default.nix:2]:
```suggestion
```

Whitespace nitpicking :).

File: pkgs/development/tools/misc/mkrom/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  src = fetchFromGitHub {
+    owner = "KnightOS";
+    repo = "mkrom";
+    rev = version;
+    sha256 = "0smilzkpf4v4w5lbx4jaah8hr24110makxjy3j92qb404pxcb5r1";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [ asciidoc libxslt.bin ];
```
Comment (danieldk) [inline pkgs/development/tools/misc/mkrom/default.nix:18]: These should be in `nativeBuildInputs`. Also, is `libxslt.bin` necessary? `asciidoc` should already have the correct path to `xsltproc`.

File: pkgs/development/tools/misc/mkrom/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+  installPhase = ''
+    mkdir -p $out/bin
+    mv bin/mkrom $out/bin
+  '';
```
Comment (danieldk) [inline pkgs/development/tools/misc/mkrom/default.nix:23]: `cmake` has install stanzas. Did you try without `installPhase`? Removing this should also properly install the man pages.

File: pkgs/development/tools/misc/mkrom/default.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+  hardeningDisable = [ "format" ];
```
Comment (danieldk) [inline pkgs/development/tools/misc/mkrom/default.nix:25]: Did you test without this?

File: pkgs/development/tools/misc/mkrom/default.nix [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+  meta = with stdenv.lib; {
+    homepage    = "https://knightos.org/";
+    description = "Combines page dumps into ROM files";
```
Comment (danieldk) [inline pkgs/development/tools/misc/mkrom/default.nix:29]: The description is a bit incomplete. What kind of ROM files?

**Review by danieldk — CHANGES_REQUESTED** [review]
The build currently fails with:

```
  a2x: ERROR: "xsltproc"  --stringparam callout.graphics 0 --stringparam navig.graphics 0 --stringparam admon.textlabel 1 --stringparam admon.graphics 0  "/nix/store/wgiyabangmawlpjw3kkr0qlbgyl3qwrf-asciidoc-8.6.9/etc/asciidoc/docbook-xsl/manpage.xsl" "/build/source/mkrom.1.xml" returned non-zero exit status 5
```

**Review by siraben — COMMENTED** [review]
Apply suggestions

**Comment by siraben** [comment]
OK, testing on NixOS as well as darwin.

File: pkgs/development/tools/misc/mkrom/default.nix (hunk 5 above)
Comment (siraben) [reply pkgs/development/tools/misc/mkrom/default.nix:23]: Yes, and it failed on darwin.

File: pkgs/development/tools/misc/mkrom/default.nix (hunk 4 above)
Comment (siraben) [reply pkgs/development/tools/misc/mkrom/default.nix:22]: I tried removing `installPhase` and the build failed.

**Comment by siraben** [comment]
NixOS build works, but `installPhase` should be adjusted for merge.

File: pkgs/development/tools/misc/mkrom/default.nix (hunk 3 above)
Comment (siraben) [reply pkgs/development/tools/misc/mkrom/default.nix]: Yes, `libxslt.bin` seems to be necessary.

**Review by danieldk — APPROVED** [review]
Result of `nixpkgs-review pr 95842` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- mkrom
</details>

**Pull request merged**